### PR TITLE
Add new platform for the TI K3 class of SoCs

### PR DESCRIPTION
**Pull request by glneo**
Add platform 'k3' for the TI K3 family. These are ARMv8 devices
and are quite different from our line of existing ARMv7 OMAP style
SoCs, hence the new platform.

Thanks,
Andrew

### CODE REVIEW

File: core/arch/arm/plat-k3/conf.mk [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ifeq ($(CFG_ARM64_core),y)
+$(call force,CFG_WITH_LPAE,y)
+ta-targets += ta_arm64
+else
+$(call force,CFG_ARM32_core,y)
+endif
+
+include core/arch/arm/cpu/cortex-armv8-0.mk
+
+arm32-platform-cflags += -Wno-error=cast-align
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-k3/conf.mk]: Why is this needed? Other ARMv8-A platforms seems to manage without it.

File: core/arch/arm/plat-k3/platform_config.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * Assumes that either TZSRAM isn't large enough or TZSRAM doesn't exist,
+ * everything is in TZDRAM.
+ * +--------+---------+
+ * |  DRAM  |  SHMEM  |
+ * +--------+---------+
+ * |        | TA_RAM  |
+ * | TZDRAM +---------+
+ * |        | TEE_RAM |
+ * +--------+---------+
+ */
```
Comment (Amit-Radur) [inline core/arch/arm/plat-k3/platform_config.h]: Shouldn't this be done via generic RAM layout( mm/generic_ram_layout.h) ?

File: core/arch/arm/plat-k3/conf.mk (hunk 1 above)
Comment (glneo) [reply core/arch/arm/plat-k3/conf.mk]: This must have been left over from some debug build, it does not seem needed anymore, will remove.

File: core/arch/arm/plat-k3/conf.mk [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+$(call force,CFG_HWSUPP_MEM_PERM_PXN,y)
+$(call force,CFG_PM_STUBS,y)
+$(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
+$(call force,CFG_WITH_ARM_TRUSTED_FW,y)
+$(call force,CFG_GIC,y)
+$(call force,CFG_ARM_GICV3,y)
+$(call force,CFG_CORE_CLUSTER_SHIFT,1)
+
+ta-targets = ta_arm32
+
```
[the remark this answers is not in the document]
Comment (glneo) [reply core/arch/arm/plat-k3/conf.mk]: Yes.

I'm not sure but I thought 32bit TAs could run under 64bit kernels. I only see one platform (rcar) that does not enable this for all builds.

[the remark this answers is not in the document]
Comment (jforissier) [reply core/arch/arm/plat-k3/conf.mk]: > I'm not sure but I thought 32bit TAs could run under 64bit kernels.

Correct (and the opposite is not true).

**Comment by glneo** [comment]
gentle ping

**Comment by glneo** [comment]
Sure, updated.

File: core/arch/arm/plat-k3/conf.mk (hunk 3 above)
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/arch/arm/plat-k3/conf.mk]: This looks alright to me

**Comment by jbech-linaro** [comment]
@glneo would you mind adding Jens A-B tag also, then we can just merge it (I've just talked to Jerome).

**Comment by glneo** [comment]
Done.

**Comment by jbech-linaro** [comment]
@glneo thanks! Consider this merged!
